search-api/src/main/scala/no/ndla/searchapi/integration/TaxonomyApiClient.scala

```diff
@@ -34,63 +34,93 @@ trait TaxonomyApiClient {
-      get[List[ResourceResourceTypeConnection]](s"$TaxonomyApiEndpoint/resource-resourcetypes/").map(_.distinct)
+    def getAllTopicSubtopicConnections(shouldUsePublishedTax: Boolean): Try[List[TopicSubtopicConnection]] =
+      get[List[TopicSubtopicConnection]](
+        s"$TaxonomyApiEndpoint/topic-subtopics/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      )
 
-    def getAllSubjectTopicConnections: Try[List[SubjectTopicConnection]] =
-      get[List[SubjectTopicConnection]](s"$TaxonomyApiEndpoint/subject-topics/")
+    def getAllResourceResourceTypeConnections(
+        shouldUsePublishedTax: Boolean
+    ): Try[List[ResourceResourceTypeConnection]] =
+      get[List[ResourceResourceTypeConnection]](
+        s"$TaxonomyApiEndpoint/resource-resourcetypes/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      ).map(_.distinct)
+
+    def getAllSubjectTopicConnections(shouldUsePublishedTax: Boolean): Try[List[SubjectTopicConnection]] =
+      get[List[SubjectTopicConnection]](
+        s"$TaxonomyApiEndpoint/subject-topics/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      )
 
-    def getAllRelevances: Try[List[Relevance]] =
-      get[List[Relevance]](s"$TaxonomyApiEndpoint/relevances/").map(_.distinct)
+    def getAllRelevances(shouldUsePublishedTax: Boolean): Try[List[Relevance]] =
+      get[List[Relevance]](s"$TaxonomyApiEndpoint/relevances/", headers = getVersionHashHeader(shouldUsePublishedTax))
+        .map(_.distinct)
 
-    def getSearchableTaxonomy(contentUri: String, filterVisibles: Boolean): Try[List[SearchableTaxonomyContext]] = {
+    def getSearchableTaxonomy(
+        contentUri: String,
+        filterVisibles: Boolean,
+        shouldUsePublishedTax: Boolean
+    ): Try[List[SearchableTaxonomyContext]] = {
       implicit val formats = SearchableLanguageFormats.JSonFormatsWithMillis
       get[List[SearchableTaxonomyContext]](
         s"$TaxonomyApiEndpoint/queries/$contentUri",
-        "filterVisibles" -> filterVisibles.toString
+        headers = getVersionHashHeader(shouldUsePublishedTax) + Map("filterVisibles" -> filterVisibles.toString)
       )
     }
 
-    val getTaxonomyBundle: Memoize[Try[TaxonomyBundle]] = Memoize(() => getTaxonomyBundleUncached)
+    private def getVersionHashHeader(shouldUsePublishedTax: Boolean): Map[String, String] = {
+      if (shouldUsePublishedTax) Map.empty else Map("versionHash" -> "default")
+    }
+
+    val getTaxonomyBundle: Memoize[Boolean, Try[TaxonomyBundle]] =
+      new Memoize(1000 * 60, shouldUsePublishedTax => getTaxonomyBundleUncached(shouldUsePublishedTax))
 
     /** The memoized function of this [[getTaxonomyBundle]] should probably be used in most cases */
-    private def getTaxonomyBundleUncached: Try[TaxonomyBundle] = {
+    private def getTaxonomyBundleUncached(shouldUsePublishedTax: Boolean): Try[TaxonomyBundle] = {
       logger.info("Fetching taxonomy in bulk...")
```

Comment:
Kunne lagt til published eller draft her kanskje.
```suggestion
      logger.info(f"Fetching ${if (shouldUsePublishedTax) "published" else "draft"} taxonomy in bulk...")
```